tasks/install-Debian.yml

```diff
@@ -1,9 +1,33 @@
 ---
 
-- name: Add apt-key
+- name: See if SNI gets in the way
+  # See https://github.com/ansible/ansible-modules-core/issues/1716
+  shell: echo | openssl s_client -connect packages.treasuredata.com:443
+  changed_when: false
+  failed_when: false
+  register: register_openssl_s_client
+  always_run: yes
+
+- name: Fix SNI issue
+  command: curl -o /etc/apt/GPG-KEY-td-agent https://packages.treasuredata.com/GPG-KEY-td-agent
+  args:
+    creates: /etc/apt/GPG-KEY-td-agent
+  when:
+    - register_openssl_s_client.rc != 0
+
+- name: Add apt-key (from file)
+  apt_key:
+    file: /etc/apt/GPG-KEY-td-agent
+    state: present
+  when:
+    - register_openssl_s_client.rc != 0
+
+- name: Add apt-key (from URL)
   apt_key:
     url: https://packages.treasuredata.com/GPG-KEY-td-agent
     state: present
+  when:
+    - register_openssl_s_client.rc == 0
```

Comment:
this seems not happen because packages.treasuredata.com uses SNI.
ansible_version does not work to know the ansible module support SNI or not?